gmso/external/convert_hoomd.py

```diff
@@ -139,17 +171,20 @@ def to_gsd_snapshot(
         u.unyt_array([lx, ly, lz]),
     )
     if parse_special_pairs:
-        _parse_pairs_information(gsd_snapshot, top)
+        _parse_pairs_information(gsd_snapshot, top, n_rigid)
     if top.n_bonds > 0:
-        _parse_bond_information(gsd_snapshot, top)
+        _parse_bond_information(gsd_snapshot, top, n_rigid)
     if top.n_angles > 0:
-        _parse_angle_information(gsd_snapshot, top)
+        _parse_angle_information(gsd_snapshot, top, n_rigid)
     if top.n_dihedrals > 0:
-        _parse_dihedral_information(gsd_snapshot, top)
+        _parse_dihedral_information(gsd_snapshot, top, n_rigid)
     if top.n_impropers > 0:
-        _parse_improper_information(gsd_snapshot, top)
+        _parse_improper_information(gsd_snapshot, top, n_rigid)
 
-    return gsd_snapshot, base_units
+    if rigid_info:
+        return gsd_snapshot, base_units, rigid_info
```

Comment:
CodeQL Pep8 doesn't like this. Would this be preferred at all:
```suggestion
        return ((gsd_snapshot, rigid_info), base_units)
```
I'm still not opposed to always returning None, if rigid_info is None. Regardless, I think we can tackle this problem down the road, if we redo the HOOMD writers and put them into a single pacakge, instead of keeping them as separate functions that cause a lot of validation headaches between the snapshot and forcefield.

Reply:
I'm not opposed either, other than the fact that it would be a breaking change for any script/program that uses the methods in `convert_hoomd`. I say we save the breaking changes for when we implement a wrapper over these methods, like we've discussed before. We should probably make a PR soon with some deprecation warnings. Maybe this should fall into a larger discussion about a GMSO 1.0 release, similar to what we did with mBuild.

Reply:
Let's be sure to raise this issue, maybe we could make a new GMSO issue for breaking changes that we should start collating.